homeassistant/components/zwave_js/discovery.py

```diff
@@ -430,13 +460,10 @@ def async_discover_values(node: ZwaveNode) -> Generator[ZwaveDiscoveryInfo, None
                 continue
             # check additional required values
             if schema.required_values is not None:
-                required_values_present = True
-                for val_scheme in schema.required_values:
-                    for val in node.values.values():
-                        if not check_value(val, val_scheme):
-                            required_values_present = False
-                            break
```

Comment:
This was returning False for the very first value that didn't match, even if the value matched later on.